tsdb/compact.go

```diff
@@ -684,6 +691,8 @@ func (c *LeveledCompactor) populateBlock(blocks []BlockReader, meta *BlockMeta,
 	c.metrics.populatingBlocks.Set(1)
 
 	globalMaxt := blocks[0].Meta().MaxTime
+	g, _ := errgroup.WithContext(c.ctx)
+	g.SetLimit(8)
```

Comment:
Let's remove the concurrency change in this pr.
It makes sense for Cortex/Thanos to use more cores during compaction since they have dedicated compactor components. But Prometheus needs CPUs for different work like scrape, query, rules, TSDB, etc so this might not be something makes sense to Prometheus.